src/ui/components/FeatureConfigPage.tsx

```diff
@@ -28,32 +30,52 @@ const FeatureConfigPage: FC = () => {
     [],
   );
 
-  const handleEnrollClick = useCallback(async () => {
-    if (selectedFeatureId === "") {
-      addToast({ message: "Invalid Input: Select feature", variant: "danger" });
-    } else if (jsonInput === "") {
-      addToast({ message: "Invalid Input: Enter JSON", variant: "danger" });
-    } else {
-      try {
-        const result = await browser.experiments.nimbus.enrollWithFeatureConfig(
-          selectedFeatureId,
-          JSON.parse(jsonInput) as object,
-          isRollout,
-        );
-
-        if (result) {
-          addToast({ message: "Enrollment successful", variant: "success" });
-        } else {
-          addToast({ message: "Enrollment failed", variant: "danger" });
-        }
-      } catch (error) {
+  const handleEnrollClick = useCallback(
+    async (forceEnroll = false) => {
+      if (selectedFeatureId === "") {
         addToast({
-          message: `Error enrolling into experiment: ${(error as Error).message ?? String(error)}`,
+          message: "Invalid Input: Select feature",
           variant: "danger",
         });
+      } else if (jsonInput === "") {
+        addToast({ message: "Invalid Input: Enter JSON", variant: "danger" });
+      } else {
+        try {
+          const result =
+            await browser.experiments.nimbus.enrollWithFeatureConfig(
+              selectedFeatureId,
+              JSON.parse(jsonInput) as object,
+              isRollout,
+              forceEnroll,
+            );
+
+          if (result.enrolled) {
+            addToast({ message: "Enrollment successful", variant: "success" });
+          } else if (result.error) {
+            setModalMessage(
+              "You are already enrolled into this feature. Would you like to force enroll by removing the previous enrollment and re-enrolling with the new configuration?",
+            );
+            setShowModal(true);
+          }
+        } catch (error) {
+          addToast({
+            message: `Error enrolling into experiment: ${(error as Error).message ?? String(error)}`,
+            variant: "danger",
+          });
+        }
       }
-    }
-  }, [jsonInput, selectedFeatureId, isRollout, addToast]);
+    },
+    [jsonInput, selectedFeatureId, isRollout, addToast],
+  );
+
+  const handleModalConfirm = useCallback(async () => {
+    setShowModal(false);
+    await handleEnrollClick(true);
+  }, [handleEnrollClick]);
```

Comment:
this needs to depend on `setShowModal` (or `setEnrollError` based on my prev comments)